pkg/controller.v2/trainjob_controller.go

```diff
@@ -15,3 +15,46 @@ limitations under the License.
 */
 
 package controllerv2
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubeflowv2 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1"
+)
+
+type TrainJobReconciler struct {
+	log      logr.Logger
+	client   client.Client
+	recorder record.EventRecorder
+}
+
+func NewTrainJobReconciler(client client.Client, recorder record.EventRecorder) *TrainJobReconciler {
+	return &TrainJobReconciler{
+		log:      ctrl.Log.WithName("trainjob-reconciler"),
```

Comment:
Should we call this logger: `trainjob-controller` ?

Reply:
I think the "train job-reconciler" would be better since the controller contains reconcilers, event handlers, and predicators.

Reply:
Yeah, but from the logging perspective we are not going to separate it, isn't ?
It might be easier from user perspective to see those logs under `trainjob-controller` name.

Reply:
Uhm, it seems that you have a point.
So, I will replace `trainjob-reconciler` with `trainjob-controller`. Thanks

Reply:
Done.